binary_transparency/firmware/cmd/ft_monitor/impl/ft_monitor.go

```diff
@@ -48,12 +50,16 @@ type MonitorOpts struct {
 	Keyword      string
 	Matched      MatchFunc
 	Annotate     bool
+	StateFile    string
 }
 
 func Main(ctx context.Context, opts MonitorOpts) error {
 	if len(opts.LogURL) == 0 {
 		return errors.New("log URL is required")
 	}
+	if len(opts.StateFile) == 0 {
```

Comment:
Is this check not contradictory to check on Line 76? If the StateFile is not provided, perhaps we should continue, as we have a mechanism to continue, in absence of a StateFile ?

Reply:
I think it would be unusual to want to operator a monitor without a state _filename_ configured - it'd be vulnerable to split views during any point at which the monitor was not running.

The section around L77 deals with the edge case of the first time the monitor is run (post this PR) and the file doesn't _yet_ exist (but will shortly be created).

Reply:
I initially coded this as optional, but then came down hard on the opinion that Al has laid out; any monitor worth its salt must absolutely be persisting its golden checkpoint.